### ISSUE TITLE
Why require v0.6.0 of BinDeps?

### ISSUE THREAD

**Issue by jpfairbanks** [JuliaIO/HDF5.jl#439]
There is a version v0.7.0 of BinDeps, is there a reason to hold on to 0.6.0?
@tkelman, do you have guidance here?

This causes a deprecation warning fixed in https://github.com/JuliaLang/BinDeps.jl/commit/e0ccae47556df91ee28f18c8fda7697d5c5eae9c:
```
INFO: Building HDF5
WARNING: Compat.ASCIIString is deprecated, use String instead.
  likely near /home/user/.julia/v0.6/HDF5/deps/build.jl:39
```

**Comment by tkelman** [comment]
If you're referring to https://github.com/JuliaIO/HDF5.jl/blob/e9092998c6661db6b4f2e3f0e4718034b0fb3d11/REQUIRE#L2, that's a lower bound. Version 0.7.0 of BinDeps is actually kind of broken on Linux on release versions of Julia, so it's been made julia-0.7-only on Linux for now. It needs a fix within BinDeps then there should be an 0.7.1 of BinDeps before too long.

**Comment by jpfairbanks** [comment]
Yes that was what I was referring to. I had been using v0.7.0 of bindeps and it was downgraded when I did the most recent `Pkg.update()`. I guess I hadn't noticed any brokenness on linux.
